pkg/controller/useraccount/useraccount_controller_test.go

```diff
@@ -749,6 +777,65 @@ func TestReconcile(t *testing.T) {
 		require.NoError(t, err)
 		require.True(t, util.HasFinalizer(userAcc, toolchainv1alpha1.FinalizerName))
 	})
+	// delete Che user fails
+	t.Run("delete che user fails", func(t *testing.T) {
+		// given
+
+		// when the member operator secret exists and has a che admin user configured then che user deletion is enabled
+		restore := test.SetEnvVarsAndRestore(t,
+			test.Env("WATCH_NAMESPACE", "toolchain-member"),
+			test.Env("MEMBER_OPERATOR_SECRET_NAME", "test-secret"),
+			// Che has two different routes for Che & Keycloak but CRW uses a single route for both.
+			// For tests we'll assume they're two different routes.
+			test.Env(configuration.MemberEnvPrefix+"_CHE_KEYCLOAK_ROUTE_NAME", "keycloak"),
+		)
+		defer restore()
+
+		defer gock.OffAll()
+		gockTokenSuccess()
+		gockFindUserNoBody("johnsmith", 400)
```

Comment:
so the deletion should fail because Che returns a `400` response when our controller sends a request to get the user info?

Reply:
That's correct, I updated the test description and added a comment to make it a bit more clear. Thanks for the question!

Reply:
ok, thanks for your answer

Reply:
actually, I thought you mention a `500` error when the user's workspaces are still running, so I was a bit surprised to see a `400` response here, but I probably misunderstood the response code when we discussed about it last time (or I just assumed it was a `500` error 😜)

Reply:
Oh, I think you're referring to the 500 error from the DELETE API. This test is assuming the Find user API fails with an error. Don't think it matters what error as long as it's not 404 which would trigger a different code path than desired.

Reply:
yes, I was probably mixing stuff. But as long as both/all cases are covered, we're fine!